### PR TITLE
Create Umi-AI.json

### PR DESCRIPTION
**Pull request by w-e-w**
## Info 
request to add https://github.com/Klokinator/Umi-AI.git to the index from https://github.com/AUTOMATIC1111/stable-diffusion-webui-extensions/issues/183

I'm currently a bit apprehensive to add this to index as this contains 900MB of TI embeddings
and including `.git` is in total taking up nearly to 2GB
and apparently there is an discussion on this topic https://github.com/Klokinator/Umi-AI/issues/18
thers talk about a 2.0 update that we use LFS or possibly some other methods

if this is true I'm inclined to not add this extension at the state and wait for a 2.0 version
## Checklist:

- [x] I have read the [`Readme.md`](https://github.com/AUTOMATIC1111/stable-diffusion-webui-extensions)
- [x] The description is written in English.
- [x] The `index.json` and `extension_template.json` have not been modified.
- [x] The `entry` is placed in the `extensions` directory with the `.json` file extension.

### CODE REVIEW

**Comment by w-e-w** [comment]
https://github.com/Tsukreya/Umi-AI-debloat

**Comment by Klokinator** [comment]
Hi, @w-e-w. The reason this extension is so massive is because originally I planned to make it a repository for all the Embeddings on the internet. As I later learned, Embeddings are (somewhat) model-specific, which presented a problem. Then the next problem is that the preview files for those embeddings are humongous.

In the beginning, there was no way to download embeddings from Civitai quickly, nor was there a way to download previews for them, either, but both of those are possible thanks to Civitai. I've been debating eradicating the Embeds for a while now, but this post gave me the motivation to do so. I recently lost my entire WebUI install due to a hard drive failure, so I might as well start from scratch and clean up shop.

I'll ping you in about a week after I do some housecleaning. I need to work on a complete revamp of Umi, but eliminating all of the Embeds and previews should bring it back down to a much more manageable size.

**Comment by w-e-w** [comment]
normally you should never use git to store binary files it is a source control system and not a web bakup cloud storage
for repositories that actually need such sort of version system with large binaries you should use [LFS (Large File Storage)](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-git-large-file-storage)
in your case you're really only using it as archive and not making changes but still it is no excuse not using LFS for large amounts of non-code files

it is best practice to separate code from assets (exception can be made when you only have a very "small" amount of assets and are "never" going to be changed)

I'm not saying having a database that you can just download for lots of embeddings is not useful
I'm just saying that this is not the way to do it

[About large files on GitHub](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github)

**Comment by Klokinator** [comment]
Embeddings are less than 100KB. The image previews I was referring to are 2MB each, on average.

I'm just going to assume you haven't looked at the internal structure of Umi AI/don't know what embeddings are, because they're very small. There's just a lot of them.

**Comment by w-e-w** [comment]
yes I'm aware that embeddings are very small 
when I say in embeddings I mean the entire folder
I was referring to embeddings and images as a whole
but even saying that 100 KB is still a lot in terms of git

for reference the entire [stable-diffusion-webui](https://github.com/AUTOMATIC1111/stable-diffusion-webui) is just 3.34MB
without history in `.git`, and with history it's still only 36.3MB, the largest files in the entire repository is a couple of images

image

if you must store stuff like that use lfs or Implement you own system to download from a external cloud storage
git and github it is not designed for storing binaries

**Comment by w-e-w** [comment]
https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases
I'm not saying that you should do it but reading this theoretically it is possible to put all your embeddings into a release page
you can have your extension download the embeddings from the release page when needed
maybe you can make it opt-in option, people who wants to embedding can click a button of some kind and all the bedrooms will be downloaded from your release page

**Comment by AshtakaOOf** [comment]
It would be better to archive the embeddings on Huggingface and have Umi-AI download them on demand.

**Comment by w-e-w** [comment]
huggingface is also a better option
but one potential issue is that I believe huggingface is blocked in China and I'm not sure if there is a mirror site
which means that if it's hosted on huggingface if a user is from China they might need a VPN

I mentioned this possible option is that it doesn't leave Github, and I believe the more mirrors of GitHub
which makes Github more accessible for users in China

**Comment by Klokinator** [comment]
I'm just going to delete all of the embeds. Already did in my version. I just need to make a push within a few days or so to clean some other stuff up.

**Comment by Klokinator** [comment]
Embeddings would be worth keeping if they weren't so "model-locked". They served a useful purpose. But with SDXL out, all the Embeds for SD1.5 will serve no purpose once SDXL (or some other model) becomes the new standard.

I'll just have to redo the entire wildcard structure to take advantage of the tweks you've added, Catwars.

**Comment by w-e-w** [comment]
@ctwrs 
> Unless the extension manager script thingy does a checkout with --depth 1, then it's not really an issue

that is actually the case since https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/10564

**Comment by Tsukreya** [comment]
> TL;DR there is no need of force push rewriting history

The reason for my fork was just because the repo was being pulled by a google colab I was using, so purging the embeddings and history did provide a speedup, but I know it's usually a bad and unnecessary idea to mess with history like that.

**Comment by w-e-w** [comment]
> > TL;DR there is no need of force push rewriting history
> 
> The reason for my fork was just because the repo was being pulled by a google colab I was using, so purging the embeddings and history did provide a speedup, but I know it's usually a bad and unnecessary idea to mess with history like that.

if you clone with `--filter=blob:none` or `--depth 1` the data in the history will not be cloned

**Comment by w-e-w** [comment]
a advice from @mcmonkey4eva
> we cannot link to a repo that has nsfw in it, much less nsfw of irl people. GitHub nuked auto for that once before and might not be so kind about reinstating access on a second offense

so it might be dangerous for us to link a project that contains NSFW content
it seems like we cannot accept this into even on the index unless NSFW content is completely purged from history
I think Force pushing might be enough but I'm not sure

**Comment by Klokinator** [comment]
I talked it over with Catwars and here's what I'm gonna do.

Current Umi AI will become Umi AI Embeds.   
[https://github.com/Klokinator/Umi-AI-Embeds](https://github.com/Klokinator/Umi-AI-Embeds)

I'll be splitting Wildcards off into their own as:  
[https://github.com/Klokinator/Umi-AI-Wildcards](https://github.com/Klokinator/Umi-AI-Wildcards)

I've already done this, but I still need to, y'know, update stuff. We need to fix some script issues and etc. Will take a bit. I need to completely redo the entire structure of the wildcards as per my V2.0 plans. Unfortunately, I had a major hard drive failure and lost my WIP build I was working on just a week ago, so I have to rebuild it all. That will take time... but hopefully not too much time. We'll see.

**Comment by w-e-w** [comment]
> I talked it over with Catwars and here's what I'm gonna do.
> 
> Current Umi AI will become Umi AI Embeds.  https://github.com/Klokinator/Umi-AI-Embeds
> 
> I'll be splitting Wildcards off into their own as: https://github.com/Klokinator/Umi-AI-Wildcards
> 
> I've already done this, but I still need to, y'know, update stuff. We need to fix some script issues and etc. Will take a bit. I need to completely redo the entire structure of the wildcards as per my V2.0 plans. Unfortunately, I had a major hard drive failure and lost my WIP build I was working on just a week ago, so I have to rebuild it all. That will take time... but hopefully not too much time. We'll see.

@Klokinator
thanks for your work
I will close this PR as this is now invalid
when you're ready to have them listed
open a new PR with the updated URL and descriptions
